File: leetcode 128/leetcode128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        setnums = set(nums)
        best = 0
        for i in setnums:
            if i - 1 not in setnums:
                length = 1
                while i + length in setnums:
                    length += 1
                best = max(best, length)
        return best

File: leetcode 128/leetcode128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        unordered_set<int> setnums(nums.begin(),nums.end());
        int best=0;
        for (int i:setnums){
            if (setnums.find(i - 1) == setnums.end()) {
                int length=1;
                while (setnums.find(i+length)!=setnums.end()) length++;
                best=max(best,length);
            }    
        }
    return best;    
    }
};
